File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-shared-5-4/IntermediaryInformation.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class IntermediaryInformation : ISerialized
    {
        public IntermediaryInformation(XmlNode xmlNode)
        {
            XmlNode routingIdsNode = xmlNode.SelectSingleNode("routingIds");
            
            if (routingIdsNode != null)
            {
                if (routingIdsNode.Attributes["href"] != null || routingIdsNode.Attributes["id"] != null) 
                {
                    if (routingIdsNode.Attributes["id"] != null) 
                    {
                        routingIdsIDRef_ = routingIdsNode.Attributes["id"].Value;
                        RoutingIds ob = new RoutingIds(routingIdsNode);
                        IDManager.SetID(routingIdsIDRef_, ob);
                    }
                    else if (routingIdsNode.Attributes["href"] != null)
                    {
                        routingIdsIDRef_ = routingIdsNode.Attributes["href"].Value;
                    }
                    else
                    {
                        routingIds_ = new RoutingIds(routingIdsNode);
                    }
                }
                else
                {
                    routingIds_ = new RoutingIds(routingIdsNode);
                }
            }
            
        
            XmlNode routingExplicitDetailsNode = xmlNode.SelectSingleNode("routingExplicitDetails");
            
            if (routingExplicitDetailsNode != null)
            {
                if (routingExplicitDetailsNode.Attributes["href"] != null || routingExplicitDetailsNode.Attributes["id"] != null) 
                {
                    if (routingExplicitDetailsNode.Attributes["id"] != null) 
                    {
                        routingExplicitDetailsIDRef_ = routingExplicitDetailsNode.Attributes["id"].Value;
                        RoutingExplicitDetails ob = new RoutingExplicitDetails(routingExplicitDetailsNode);
                        IDManager.SetID(routingExplicitDetailsIDRef_, ob);
                    }
                    else if (routingExplicitDetailsNode.Attributes["href"] != null)
                    {
                        routingExplicitDetailsIDRef_ = routingExplicitDetailsNode.Attributes["href"].Value;
                    }
                    else
                    {
                        routingExplicitDetails_ = new RoutingExplicitDetails(routingExplicitDetailsNode);
                    }
                }
                else
                {
                    routingExplicitDetails_ = new RoutingExplicitDetails(routingExplicitDetailsNode);
                }
            }
            
        
            XmlNode routingIdsAndExplicitDetailsNode = xmlNode.SelectSingleNode("routingIdsAndExplicitDetails");
            
            if (routingIdsAndExplicitDetailsNode != null)
            {
                if (routingIdsAndExplicitDetailsNode.Attributes["href"] != null || routingIdsAndExplicitDetailsNode.Attributes["id"] != null) 
                {
                    if (routingIdsAndExplicitDetailsNode.Attributes["id"] != null) 
                    {
                        routingIdsAndExplicitDetailsIDRef_ = routingIdsAndExplicitDetailsNode.Attributes["id"].Value;
                        RoutingIdsAndExplicitDetails ob = new RoutingIdsAndExplicitDetails(routingIdsAndExplicitDetailsNode);
                        IDManager.SetID(routingIdsAndExplicitDetailsIDRef_, ob);
                    }
                    else if (routingIdsAndExplicitDetailsNode.Attributes["href"] != null)
                    {
                        routingIdsAndExplicitDetailsIDRef_ = routingIdsAndExplicitDetailsNode.Attributes["href"].Value;
                    }
                    else
                    {
                        routingIdsAndExplicitDetails_ = new RoutingIdsAndExplicitDetails(routingIdsAndExplicitDetailsNode);
                    }
                }
                else
                {
                    routingIdsAndExplicitDetails_ = new RoutingIdsAndExplicitDetails(routingIdsAndExplicitDetailsNode);
                }
            }
            
        
            XmlNode intermediarySequenceNumberNode = xmlNode.SelectSingleNode("intermediarySequenceNumber");
            
            if (intermediarySequenceNumberNode != null)
            {
                if (intermediarySequenceNumberNode.Attributes["href"] != null || intermediarySequenceNumberNode.Attributes["id"] != null) 
                {
                    if (intermediarySequenceNumberNode.Attributes["id"] != null) 
                    {
                        intermediarySequenceNumberIDRef_ = intermediarySequenceNumberNode.Attributes["id"].Value;
                        XsdTypePositiveInteger ob = new XsdTypePositiveInteger(intermediarySequenceNumberNode);
                        IDManager.SetID(intermediarySequenceNumberIDRef_, ob);
                    }
                    else if (intermediarySequenceNumberNode.Attributes["href"] != null)
                    {
                        intermediarySequenceNumberIDRef_ = intermediarySequenceNumberNode.Attributes["href"].Value;
                    }
                    else
                    {
                        intermediarySequenceNumber_ = new XsdTypePositiveInteger(intermediarySequenceNumberNode);
                    }
                }
                else
                {
                    intermediarySequenceNumber_ = new XsdTypePositiveInteger(intermediarySequenceNumberNode);
                }
            }
            
        
            XmlNode intermediaryPartyReferenceNode = xmlNode.SelectSingleNode("intermediaryPartyReference");
            
            if (intermediaryPartyReferenceNode != null)
            {
                if (intermediaryPartyReferenceNode.Attributes["href"] != null || intermediaryPartyReferenceNode.Attributes["id"] != null) 
                {
                    if (intermediaryPartyReferenceNode.Attributes["id"] != null) 
                    {
                        intermediaryPartyReferenceIDRef_ = intermediaryPartyReferenceNode.Attributes["id"].Value;
                        PartyReference ob = new PartyReference(intermediaryPartyReferenceNode);
                        IDManager.SetID(intermediaryPartyReferenceIDRef_, ob);
                    }
                    else if (intermediaryPartyReferenceNode.Attributes["href"] != null)
                    {
                        intermediaryPartyReferenceIDRef_ = intermediaryPartyReferenceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        intermediaryPartyReference_ = new PartyReference(intermediaryPartyReferenceNode);
                    }
                }
                else
                {
                    intermediaryPartyReference_ = new PartyReference(intermediaryPartyReferenceNode);
                }
            }
            
        
        }
        
    
        #region RoutingIds_
        private RoutingIds routingIds_;
        public RoutingIds RoutingIds_
        {
            get
            {
                if (this.routingIds_ != null)
                {
                    return this.routingIds_; 
                }
                else if (this.routingIdsIDRef_ != null)
                {
                    routingIds_ = IDManager.getID(routingIdsIDRef_) as RoutingIds;
                    return this.routingIds_; 
                }
                else
                {
                      return this.routingIds_; 
                }
            }
            set
            {
                if (this.routingIds_ != value)
                {
                    this.routingIds_ = value;
                }
            }
        }
        #endregion
        
        public string routingIdsIDRef_ { get; set; }
        #region RoutingExplicitDetails_
        private RoutingExplicitDetails routingExplicitDetails_;
        public RoutingExplicitDetails RoutingExplicitDetails_
        {
            get
            {
                if (this.routingExplicitDetails_ != null)
                {
                    return this.routingExplicitDetails_; 
                }
                else if (this.routingExplicitDetailsIDRef_ != null)
                {
                    routingExplicitDetails_ = IDManager.getID(routingExplicitDetailsIDRef_) as RoutingExplicitDetails;
                    return this.routingExplicitDetails_; 
                }
                else
                {
                      return this.routingExplicitDetails_; 
                }
            }
            set
            {
                if (this.routingExplicitDetails_ != value)
                {
                    this.routingExplicitDetails_ = value;
                }
            }
        }
        #endregion
        
        public string routingExplicitDetailsIDRef_ { get; set; }
        #region RoutingIdsAndExplicitDetails_
        private RoutingIdsAndExplicitDetails routingIdsAndExplicitDetails_;
        public RoutingIdsAndExplicitDetails RoutingIdsAndExplicitDetails_
        {
            get
            {
                if (this.routingIdsAndExplicitDetails_ != null)
                {
                    return this.routingIdsAndExplicitDetails_; 
                }
                else if (this.routingIdsAndExplicitDetailsIDRef_ != null)
                {
                    routingIdsAndExplicitDetails_ = IDManager.getID(routingIdsAndExplicitDetailsIDRef_) as RoutingIdsAndExplicitDetails;
                    return this.routingIdsAndExplicitDetails_; 
                }
                else
                {
                      return this.routingIdsAndExplicitDetails_; 
                }
            }
            set
            {
                if (this.routingIdsAndExplicitDetails_ != value)
                {
                    this.routingIdsAndExplicitDetails_ = value;
                }
            }
        }
        #endregion
        
        public string routingIdsAndExplicitDetailsIDRef_ { get; set; }
        #region IntermediarySequenceNumber_
        private XsdTypePositiveInteger intermediarySequenceNumber_;
        public XsdTypePositiveInteger IntermediarySequenceNumber_
        {
            get
            {
                if (this.intermediarySequenceNumber_ != null)
                {
                    return this.intermediarySequenceNumber_; 
                }
                else if (this.intermediarySequenceNumberIDRef_ != null)
                {
                    intermediarySequenceNumber_ = IDManager.getID(intermediarySequenceNumberIDRef_) as XsdTypePositiveInteger;
                    return this.intermediarySequenceNumber_; 
                }
                else
                {
                      return this.intermediarySequenceNumber_; 
                }
            }
            set
            {
                if (this.intermediarySequenceNumber_ != value)
                {
                    this.intermediarySequenceNumber_ = value;
                }
            }
        }
        #endregion
        
        public string intermediarySequenceNumberIDRef_ { get; set; }
        #region IntermediaryPartyReference_
        private PartyReference intermediaryPartyReference_;
        public PartyReference IntermediaryPartyReference_
        {
            get
            {
                if (this.intermediaryPartyReference_ != null)
                {
                    return this.intermediaryPartyReference_; 
                }
                else if (this.intermediaryPartyReferenceIDRef_ != null)
                {
                    intermediaryPartyReference_ = IDManager.getID(intermediaryPartyReferenceIDRef_) as PartyReference;
                    return this.intermediaryPartyReference_; 
                }
                else
                {
                      return this.intermediaryPartyReference_; 
                }
            }
            set
            {
                if (this.intermediaryPartyReference_ != value)
                {
                    this.intermediaryPartyReference_ = value;
                }
            }
        }
        #endregion
        
        public string intermediaryPartyReferenceIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-shared-5-4/IntermediaryInformation.cpp
// IntermediaryInformation.cpp 
#include "IntermediaryInformation.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

IntermediaryInformation::IntermediaryInformation(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //routingIdsNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* routingIdsNode = xmlNode->FirstChildElement("routingIds");

   if(routingIdsNode){routingIdsIsNull_ = false;}
   else{routingIdsIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- routingIdsNode , address : " << routingIdsNode << std::endl;
   #endif
   if(routingIdsNode)
   {
      if(routingIdsNode->Attribute("href") || routingIdsNode->Attribute("id"))
      {
          if(routingIdsNode->Attribute("id"))
          {
             routingIdsIDRef_ = routingIdsNode->Attribute("id");
             routingIds_ = boost::shared_ptr<RoutingIds>(new RoutingIds(routingIdsNode));
             routingIds_->setID(routingIdsIDRef_);
             IDManager::instance().setID(routingIdsIDRef_,routingIds_);
          }
          else if(routingIdsNode->Attribute("href")) { routingIdsIDRef_ = routingIdsNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { routingIds_ = boost::shared_ptr<RoutingIds>(new RoutingIds(routingIdsNode));}
   }

   //routingExplicitDetailsNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* routingExplicitDetailsNode = xmlNode->FirstChildElement("routingExplicitDetails");

   if(routingExplicitDetailsNode){routingExplicitDetailsIsNull_ = false;}
   else{routingExplicitDetailsIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- routingExplicitDetailsNode , address : " << routingExplicitDetailsNode << std::endl;
   #endif
   if(routingExplicitDetailsNode)
   {
      if(routingExplicitDetailsNode->Attribute("href") || routingExplicitDetailsNode->Attribute("id"))
      {
          if(routingExplicitDetailsNode->Attribute("id"))
          {
             routingExplicitDetailsIDRef_ = routingExplicitDetailsNode->Attribute("id");
             routingExplicitDetails_ = boost::shared_ptr<RoutingExplicitDetails>(new RoutingExplicitDetails(routingExplicitDetailsNode));
             routingExplicitDetails_->setID(routingExplicitDetailsIDRef_);
             IDManager::instance().setID(routingExplicitDetailsIDRef_,routingExplicitDetails_);
          }
          else if(routingExplicitDetailsNode->Attribute("href")) { routingExplicitDetailsIDRef_ = routingExplicitDetailsNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { routingExplicitDetails_ = boost::shared_ptr<RoutingExplicitDetails>(new RoutingExplicitDetails(routingExplicitDetailsNode));}
   }

   //routingIdsAndExplicitDetailsNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* routingIdsAndExplicitDetailsNode = xmlNode->FirstChildElement("routingIdsAndExplicitDetails");

   if(routingIdsAndExplicitDetailsNode){routingIdsAndExplicitDetailsIsNull_ = false;}
   else{routingIdsAndExplicitDetailsIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- routingIdsAndExplicitDetailsNode , address : " << routingIdsAndExplicitDetailsNode << std::endl;
   #endif
   if(routingIdsAndExplicitDetailsNode)
   {
      if(routingIdsAndExplicitDetailsNode->Attribute("href") || routingIdsAndExplicitDetailsNode->Attribute("id"))
      {
          if(routingIdsAndExplicitDetailsNode->Attribute("id"))
          {
             routingIdsAndExplicitDetailsIDRef_ = routingIdsAndExplicitDetailsNode->Attribute("id");
             routingIdsAndExplicitDetails_ = boost::shared_ptr<RoutingIdsAndExplicitDetails>(new RoutingIdsAndExplicitDetails(routingIdsAndExplicitDetailsNode));
             routingIdsAndExplicitDetails_->setID(routingIdsAndExplicitDetailsIDRef_);
             IDManager::instance().setID(routingIdsAndExplicitDetailsIDRef_,routingIdsAndExplicitDetails_);
          }
          else if(routingIdsAndExplicitDetailsNode->Attribute("href")) { routingIdsAndExplicitDetailsIDRef_ = routingIdsAndExplicitDetailsNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { routingIdsAndExplicitDetails_ = boost::shared_ptr<RoutingIdsAndExplicitDetails>(new RoutingIdsAndExplicitDetails(routingIdsAndExplicitDetailsNode));}
   }

   //intermediarySequenceNumberNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* intermediarySequenceNumberNode = xmlNode->FirstChildElement("intermediarySequenceNumber");

   if(intermediarySequenceNumberNode){intermediarySequenceNumberIsNull_ = false;}
   else{intermediarySequenceNumberIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- intermediarySequenceNumberNode , address : " << intermediarySequenceNumberNode << std::endl;
   #endif
   if(intermediarySequenceNumberNode)
   {
      if(intermediarySequenceNumberNode->Attribute("href") || intermediarySequenceNumberNode->Attribute("id"))
      {
          if(intermediarySequenceNumberNode->Attribute("id"))
          {
             intermediarySequenceNumberIDRef_ = intermediarySequenceNumberNode->Attribute("id");
             intermediarySequenceNumber_ = boost::shared_ptr<XsdTypePositiveInteger>(new XsdTypePositiveInteger(intermediarySequenceNumberNode));
             intermediarySequenceNumber_->setID(intermediarySequenceNumberIDRef_);
             IDManager::instance().setID(intermediarySequenceNumberIDRef_,intermediarySequenceNumber_);
          }
          else if(intermediarySequenceNumberNode->Attribute("href")) { intermediarySequenceNumberIDRef_ = intermediarySequenceNumberNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { intermediarySequenceNumber_ = boost::shared_ptr<XsdTypePositiveInteger>(new XsdTypePositiveInteger(intermediarySequenceNumberNode));}
   }

   //intermediaryPartyReferenceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* intermediaryPartyReferenceNode = xmlNode->FirstChildElement("intermediaryPartyReference");

   if(intermediaryPartyReferenceNode){intermediaryPartyReferenceIsNull_ = false;}
   else{intermediaryPartyReferenceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- intermediaryPartyReferenceNode , address : " << intermediaryPartyReferenceNode << std::endl;
   #endif
   if(intermediaryPartyReferenceNode)
   {
      if(intermediaryPartyReferenceNode->Attribute("href") || intermediaryPartyReferenceNode->Attribute("id"))
      {
          if(intermediaryPartyReferenceNode->Attribute("id"))
          {
             intermediaryPartyReferenceIDRef_ = intermediaryPartyReferenceNode->Attribute("id");
             intermediaryPartyReference_ = boost::shared_ptr<PartyReference>(new PartyReference(intermediaryPartyReferenceNode));
             intermediaryPartyReference_->setID(intermediaryPartyReferenceIDRef_);
             IDManager::instance().setID(intermediaryPartyReferenceIDRef_,intermediaryPartyReference_);
          }
          else if(intermediaryPartyReferenceNode->Attribute("href")) { intermediaryPartyReferenceIDRef_ = intermediaryPartyReferenceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { intermediaryPartyReference_ = boost::shared_ptr<PartyReference>(new PartyReference(intermediaryPartyReferenceNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<RoutingIds> IntermediaryInformation::getRoutingIds()
{
   if(!this->routingIdsIsNull_){
        if(routingIdsIDRef_ != ""){
             return boost::shared_static_cast<RoutingIds>(IDManager::instance().getID(routingIdsIDRef_));
        }else{
             return this->routingIds_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<RoutingIds>();
   }
}
boost::shared_ptr<RoutingExplicitDetails> IntermediaryInformation::getRoutingExplicitDetails()
{
   if(!this->routingExplicitDetailsIsNull_){
        if(routingExplicitDetailsIDRef_ != ""){
             return boost::shared_static_cast<RoutingExplicitDetails>(IDManager::instance().getID(routingExplicitDetailsIDRef_));
        }else{
             return this->routingExplicitDetails_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<RoutingExplicitDetails>();
   }
}
boost::shared_ptr<RoutingIdsAndExplicitDetails> IntermediaryInformation::getRoutingIdsAndExplicitDetails()
{
   if(!this->routingIdsAndExplicitDetailsIsNull_){
        if(routingIdsAndExplicitDetailsIDRef_ != ""){
             return boost::shared_static_cast<RoutingIdsAndExplicitDetails>(IDManager::instance().getID(routingIdsAndExplicitDetailsIDRef_));
        }else{
             return this->routingIdsAndExplicitDetails_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<RoutingIdsAndExplicitDetails>();
   }
}
boost::shared_ptr<XsdTypePositiveInteger> IntermediaryInformation::getIntermediarySequenceNumber()
{
   if(!this->intermediarySequenceNumberIsNull_){
        if(intermediarySequenceNumberIDRef_ != ""){
             return boost::shared_static_cast<XsdTypePositiveInteger>(IDManager::instance().getID(intermediarySequenceNumberIDRef_));
        }else{
             return this->intermediarySequenceNumber_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypePositiveInteger>();
   }
}
boost::shared_ptr<PartyReference> IntermediaryInformation::getIntermediaryPartyReference()
{
   if(!this->intermediaryPartyReferenceIsNull_){
        if(intermediaryPartyReferenceIDRef_ != ""){
             return boost::shared_static_cast<PartyReference>(IDManager::instance().getID(intermediaryPartyReferenceIDRef_));
        }else{
             return this->intermediaryPartyReference_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<PartyReference>();
   }
}
}

